src/main/java/com/xceptance/xlt/report/providers/CustomLogsReportProvider.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2005-2024 Xceptance Software Technologies GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.xceptance.xlt.report.providers;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import com.xceptance.xlt.api.engine.Data;
+import com.xceptance.xlt.api.report.AbstractReportProvider;
+import com.xceptance.xlt.common.XltConstants;
+import com.xceptance.xlt.report.ReportGeneratorConfiguration;
+
+/**
+ * Report provider generating a report fragment about the custom data logs collected during the test run
+ */
+public class CustomLogsReportProvider extends AbstractReportProvider
+{
+
+    private static final String CUSTOM_DATA = "custom_data_logs";
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Object createReportFragment()
+    {
+        final CustomLogsReport report = new CustomLogsReport();
+
+        final Path resultsDir = ((ReportGeneratorConfiguration) getConfiguration()).getResultsDirectory().getPath();
+        final Path targetDir = Paths.get(getConfiguration().getReportDirectory() + File.separator + CUSTOM_DATA);
+    
+        CustomLogFinder finder = new CustomLogFinder();
+        
+        finder.setBaseDir(resultsDir);
+        finder.setTargetDir(targetDir);
+        
+        try
+        {
+            Files.walkFileTree(resultsDir, finder);
```

Comment:
XLT is able to create a report also from a result archive (i.e. a zip file). That's why we usually deal with FileObjects, not Paths or Files. This means we cannot use the standard file tree walker, but have to implement that piece ourselves. See DataReaderThread or DataProcessor for inspiration how to work with FileObjects. The actual tree walker is likely just a recursive method.